src/components/popover-container/popover-container.pw.tsx

```diff
@@ -0,0 +1,683 @@
+  // The fix for this bug has been released in Playwright v1.40.0.
+  // https://github.com/microsoft/playwright/pull/27711
+  test.skip("should not close when the escape key is pressed and the Select List is open", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<PopoverContainerWithSelect />);
+
+    await popoverSettingsIcon(page).click();
+    const selectTextElement = selectText(page);
+    await selectTextElement.click();
+    const selectListElement = selectListText(page, "red");
+    await waitForAnimationEnd(selectListElement);
+    await page.keyboard.press("ArrowDown");
+    await page.keyboard.press("ArrowDown");
+    await popoverContainerComponent(page).press("Escape");
+
+    await expect(selectListText(page, "green")).not.toBeVisible();
+    await expect(popoverContainerContent(page)).toBeVisible();
+  });
+
+  test("should close when the escape key is pressed with focus on the Select component", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<PopoverContainerWithSelect />);
+
+    await popoverSettingsIcon(page).click();
+    const selectTextElement = selectText(page);
+    await selectTextElement.click();
+    await selectTextElement.press("Escape");
+    const popoverContainerContentElement = popoverContainerContent(page);
+
+    await expect(popoverContainerContentElement).not.toBeVisible();
+  });
+});
+
+test.describe("Event tests", () => {
+  test("should call onOpen callback when a click event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <PopoverContainer
+        onOpen={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const popoverSettingsIconElement = popoverSettingsIcon(page);
+    await popoverSettingsIconElement.click();
+
+    await expect(callbackCount).toBe(1);
+  });
+
+  ([...keyToTrigger] as const).forEach((key) => {
+    test(`should call onOpen callback when a keyboard event is triggered by ${key} key`, async ({
+      mount,
+      page,
+    }) => {
+      let callbackCount = 0;
+      await mount(
+        <PopoverContainer
+          onOpen={() => {
+            callbackCount += 1;
+          }}
+        />
+      );
+
+      const popoverSettingsIconElement = popoverSettingsIcon(page);
+      await popoverSettingsIconElement.press(key);
+
+      await expect(callbackCount).toBe(1);
+    });
+  });
+
+  test("should call onClose callback when a click event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <PopoverContainerComponent
+        onClose={() => {
+          callbackCount += 1;
+        }}
+        open
+      />
+    );
+
+    const popoverCloseIconElement = popoverCloseIcon(page);
+    await popoverCloseIconElement.click();
+
+    await expect(callbackCount).toBe(1);
+  });
+
+  ([...keyToTrigger] as const).forEach((key) => {
+    test(`should call onClose callback when a ${key} keyboard event is triggered`, async ({
```

Comment:
```suggestion
    test(`should call onClose callback when a keyboard event is triggered by ${key} key`, async ({
```

**suggestion** (non-blocking): this is the test description you used for the `onOpen` version a couple of tests up from here, and I much prefer it. Mainly so you don't get "when a Enter keyboard event is triggered" in one of the tests 😅  - but also for consistency.

Reply:
Consistently inconsistent. I'll get it changed.